includes/admin/class-wc-rest-payments-authorizations-controller.php

```diff
@@ -58,11 +60,14 @@ public function register_routes() {
 	 * @param WP_REST_Request $request Full data about the request.
 	 */
 	public function get_authorizations( WP_REST_Request $request ) {
-		$page      = (int) $request->get_param( 'page' );
-		$page_size = (int) $request->get_param( 'pagesize' );
-		$sort      = $request->get_param( 'sort' );
-		$direction = $request->get_param( 'direction' );
-		return $this->forward_request( 'list_authorizations', [ $page, $page_size, $sort, $direction ] );
+		$wcpay_request = Paginated_Request::create();
```

Comment:
Considering that `uri`, `page`, `page_size`, `sort_by`, and `sort_direction` are universal, do you think we could create a non-final `create_from_rest_request` method within the `Paginated` class, something like this:

```php
class Paginated {
	public static function from_rest_request( $request ) {
		$wcpay_request = static::create();
		$wcpay_request->set_page( (int) $request->get_param( 'page' ) );
		$wcpay_request->set_page_size( (int) $request->get_param( 'pagesize' ) );
		$wcpay_request->set_sort_by( $request->get_param( 'sort' ) );
		$wcpay_request->set_sort_direction( $request->get_param( 'direction' ) );	
		return $wcpay_request;
	}
}
```
This way sub-classes will be able to nicely use filters as well:

```php
class List_Disputes extends Paginated {
	public static function from_rest_request( $request ) {
		$filters = [
			'match'           => $request->get_param( 'match' ),
			'currency_is'     => $request->get_param( 'store_currency_is' ),
			'created_before'  => $request->get_param( 'date_before' ),
			'created_after'   => $request->get_param( 'date_after' ),
			'created_between' => $request->get_param( 'date_between' ),
			'search'          => $request->get_param( 'search' ),
			'status_is'       => $request->get_param( 'status_is' ),
			'status_is_not'   => $request->get_param( 'status_is_not' ),
		];

		$wcpay_request = parent::from_rest_request( $request );
		$wcpay_request->set_filters( $filters );
		return $wcpay_request;
	}
}
```

By doing so we will simultaneously add a good base for a bridge between REST and server requests.

Reply:
All paginated request classes have this feature now.